cachematrix.R

```diff
@@ -1,15 +1,35 @@
-## Put comments here that give an overall description of what your
-## functions do
+## These functions generate, cache, and return the cache of the inverse
+## of a matrix.
 
-## Write a short comment describing this function
+## This function creates a "matrix" object and caches its inverse.
 
 makeCacheMatrix <- function(x = matrix()) {
-
+  m <- NULL
+  set <- function(y) {
+    x <<- y
+    m <<- NULL
+  }
+  get <- function() x
+  setsolve <- function(solve) m <<- solve
+  getsolve <- function() m
+  list(set = set, get = get,
+       setsolve = setsolve,
+       getsolve = getsolve)
```

Comment:
I have no ideas what any of this is doing, but it looks like is a touch redundant.  Not blocking, but you variable and functions names aren't very descriptive.